spec/services/attribute_asserter_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+describe AttributeAsserter do
+  include SamlAuthHelper
+
+  let(:user) do
+    user = create(:user, :signed_up)
+    user.profiles.create(
+      active: true,
+      activated_at: Time.current,
+      verified_at: Time.current,
+      first_name: 'Jane'
+    )
+    user
+  end
+  let(:service_provider) { ServiceProvider.new(sp1_saml_settings.issuer) }
+  let(:subject) { described_class.new(user, service_provider, sp1_authnrequest) }
+
+  describe '#build' do
+    before do
+      allow_any_instance_of(ServiceProvider).to receive(:attribute_bundle).and_return(
+        %w(email first_name)
+      )
+      subject.build
+    end
+
+    it 'sets user asserted_attributes' do
+      expect(user.asserted_attributes).to have_key :email
+      expect(user.asserted_attributes).to have_key :first_name
+      expect(user.asserted_attributes).to_not have_key :last_name
```

Comment:
Shouldn't the user profile have a `last_name` field for this test to be effective?